app/controllers/concerns/hyrax/works_controller_behavior.rb

```diff
@@ -55,21 +55,12 @@ def new
     end
 
     def create
-      # Caching the original input params in case the form is not valid
-      original_input_params_for_form = params[hash_key_for_curation_concern].deep_dup
-      if create_work
-        after_create_response
+      case curation_concern
+      when ActiveFedora::Base
+        original_input_params_for_form = params[hash_key_for_curation_concern].deep_dup
+        actor.create(actor_environment) ? after_create_response : after_create_error(curation_concern.errors, original_input_params_for_form)
       else
-        respond_to do |wants|
```

Comment:
The else block is error processing.  This was moved to `#after_create_error`.  Again, this matches the pattern used for Collections and represents a cleaner separation of purpose.